src/jit/codegenxarch.cpp

```diff
@@ -7049,6 +7050,12 @@ void CodeGen::genCompareLong(GenTreePtr treeNode)
     // Emit the compare instruction
     getEmitter()->emitInsBinary(ins, cmpAttr, hiOp1, hiOp2);
 
+    // If the result is not being materialized in a register, we're done.
+    if (targetReg == REG_NA)
+    {
+        return;
+    }
+
```

Comment:
As per my understanding this routine gets called in two cases: 1) when the result needs to be materialized into a register and 2) when GT_CMP result is unused.
Even in case (2), GT_CMP will not marked as contained and it will need to materialize the result into a register.

Two things:

1) why are we deleting assert on line 7039, if this routine gets called when targetReg != REG_NA.?
2) The bail-out is happening after we generate hiop compare.